Mathlib/CategoryTheory/Groupoid.lean

```diff
@@ -132,6 +146,14 @@ instance InducedCategory.groupoid {C : Type u} (D : Type u₂) [Groupoid.{v} D]
     inv_comp := fun f => Groupoid.inv_comp f
     comp_inv := fun f => Groupoid.comp_inv f }
 
+instance InducedCategory.isGroupoid {C : Type u} (D : Type u₂)
+    [Category.{v} D] [IsGroupoid D] (F : C → D) :
+    IsGroupoid (InducedCategory D F) where
+  all_is_iso {x y} f := by
+    let g : (F x ⟶ F y) := (fullyFaithfulInducedFunctor F).preimage f
+    obtain ⟨i, h, k⟩ : IsIso g := by infer_instance
+    exact ⟨i, h, k⟩
```

Comment:
This is a particular case of a lemma
```lean
lemma isGroupoid_of_reflects_iso {C D : Type*} [Category C] [Category D]
    (F : C ⥤ D) [F.ReflectsIsomorphisms] [IsGroupoid D] :
    IsGroupoid C where
  all_is_iso _ := isIso_of_reflects_iso _ F
```

Reply:
We need to import `import Mathlib.CategoryTheory.Functor.ReflectsIso` for this, but this runs into an import cycle it seems.

Reply:
(I guess we could split `CategoryTheory.Functor.ReflectsIso` into two files, one that only contains the def, and one that imports `CategoryTheory.Balanced` and `CategoryTheory.Functor.EpiMono`, but that might be work for an other PR).

Reply:
Yes, the best would be to proceed first with the splitting of `ReflectsIso`, and then add `IsGroupoid`.